src/Microsoft.AspNet.Mvc.ViewFeatures/ModelStateDictionaryExtensions.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq.Expressions;
+using Microsoft.AspNet.Mvc.ModelBinding;
+using Microsoft.AspNet.Mvc.ViewFeatures;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Extensions methods for <see cref="ModelStateDictionary"/>.
+    /// </summary>
+    public static class ModelStateDictionaryExtensions
+    {
+        /// <summary>
+        /// Adds the specified <paramref name="errorMessage"/> to the <see cref="ModelState.Errors"/> instance
+        /// that is associated with the specified <paramref name="expression"/> text key repressentation.
+        /// </summary>
+        /// <typeparam name="TModel">The type of the model.</typeparam>
+        /// <param name="modelState">The <see cref="ModelStateDictionary"/> instance this method extends.</param>
+        /// <param name="expression">The expression of <typeparamref name="TModel"/> from
+        /// which text representation will be used as a key of the <see cref="ModelState"/> to add errors to.</param>
+        /// <param name="errorMessage"></param>
```

Comment:
Don't invent new terms for this documentation, especially not "text key representation". Suggest being consistent with what we have in https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/IHtmlHelperOfT.cs That includes simply saying `specified <paramref name="expression"/>` and documenting the parameter as `An expression to be evaluated against an item in the current model.`. (Realize the expression won't be executed but it is "evaluated" in a very general sense.)